src/colorizer/AnnotationData.ts

```diff
@@ -48,39 +48,56 @@ export interface IAnnotationData {
+   * [objectIds]}}`.
    *
-   * The record's keys are an index of a label (as returned by `getLabels()`),
-   * and the values are arrays of object IDs present for that time frame that
-   * have that label applied.
+   * Each time (by frame number) maps to a record of labeled object IDs present
+   * at that time. The record's keys are a label (by index), and the value is an
+   * array of the objects (by ID) present for that time frame that have that
+   * label applied. (If no objects have a label applied at a given time, it will
+   * not be present in the record.)
+   *
+   * A dataset with a single frame and an object with ID 0, labeled with label
+   * 0, would return `{0: {0: [0]}}`, in the format `{time: {labelId:
+   * [objectIds]}}`.
    *
    * @param dataset The dataset to use for time information.
    * @returns a map from time to a record of label indices to IDs.
    * @example
    * ```typescript
-   * // Let's say we have two labels (0 and 1). There are objects with IDs 1, 2,
-   * // and 3 at time 0, and IDs 4 and 5 at time 1.
-   * // Label 0 has been applied to objects 1, 2, 3, and 4.
-   * // Label 1 has been applied to objects 3 and 5.
+   * // Let's say we have two labels (0 and 1). There are objects with IDs 11, 12,
+   * // and 13 at time 234, and ID 14 and 15 at time 577.
+   * // Label 0 has been applied to objects 11, 12, and 13.
+   * // Label 1 has been applied to objects 13, 14, and 15.
    *
    * const timeToLabelMap = getTimeToLabelIdMap(some_dataset);
-   * timeToLabelMap.get(0); // { 0: [1, 2, 3], 1: [3] }
-   * timeToLabelMap.get(1); // { 0: [4], 1: [5]}
+   * timeToLabelMap.get(234); // { 0: [11, 12, 13], 1: [13] }
+   * timeToLabelMap.get(577); // { 1: [14, 15]}
    * ```
    * */
   getTimeToLabelIdMap(dataset: Dataset): Map<number, Record<number, number[]>>;
+}
+
+export interface IAnnotationDataSetters {
+  /** Creates a new label and returns its index in the array of labels (as
+   * returned by `getLabels()`).
+   * @param name The name of the label. If no name is provided, a default name
+   * ("Label {number}") will be used.
+   * @param color The color to use for the label. If no color is provided, a
+   * color will be chosen from the default categorical palette.
+   * */
+  createNewLabel(name?: string, color?: Color): number;
 
   setLabelName(labelIdx: number, name: string): void;
   setLabelColor(labelIdx: number, color: Color): void;
   deleteLabel(labelIdx: number): void;
 
-  applyLabelToId(labelIdx: number, id: number): void;
-  removeLabelFromId(labelIdx: number, id: number): void;
-  toggleLabelOnId(labelIdx: number, id: number): void;
+  setLabelOnId(labelIdx: number, id: number, value: boolean): void;
 }
 
-export class AnnotationData implements IAnnotationData {
+export type IAnnotationData = IAnnotationDataGetters & IAnnotationDataSetters;
+
+export class AnnotationData implements IAnnotationDataGetters, IAnnotationDataSetters {
```

Comment:
Just curious: can `AnnotationData` just implement `IAnnotationData` directly? If not, would defining `IAnnotationData` as an interface like...

```typescript
export interface IAnnotationData extends IAnnotationDataGetters, IAnnotationDataSetters {}
```

...make it possible?